lib/core/connection/connection.js

```diff
@@ -271,6 +271,10 @@ class Connection extends EventEmitter {
 
       // Iterate over all buffers and write them in order to the socket
       for (let i = 0; i < buffer.length; i++) {
+        this.socket.on('error', function(){
+          console.log("CLIENT HAS ERRORED");
+          break;
+        });
```

Comment:
This will add an error listener per buffer in the BufferList passed to write and the break on line 276 won't have an effect on the synchronous for loop as the socket can emit the error event at a time later than when the loop is run.

Reply:
I tried this on my fork and it's not accomplishing what I want. I'm going to close this.